### ISSUE TITLE
Don't understand how to use relationships

### ISSUE THREAD

**Issue by tomq42** [derekdowling/go-json-spec-handler#27]
There seem to be very few real examples in the code (yes, I have looked at the tests, but it's hard to tell what's real use, and that's testing internal structures).

I was expecting some way of designating that this field in this struct should be represented as a link in the JSON. I'm not sure I understand how linked resources are intended to be used with this code.

Any help please?
Thanks.

**Comment by derekdowling** [comment]
My apologies. Some of the more complex examples (links, metadata, relationships) I haven't written a ton of documentation around and have accepted a bunch of contributions to round that functionality out. Can you link me to what exactly you are trying to do within the [JSONAPI specification](http://jsonapi.org/format/) so I can better answer your question or turn this into a trackable issue to improve? Thanks!

**Comment by derekdowling** [comment]
@tomq42 did that solve your problem? I'd like to resolve this issue.

**Comment by tomq42** [comment]
Yes, I did figure that out.
What I realised was that json api prohibits relationships at lower levels in the
"body". So serialising the body is a straightforward golang json task, it
doesn't need any special annotations and so on. The relationships can only exist
at the top level, and have to be created manually.
What I wanted was to have a relationship down in a nested object somewhere. I
had a nested object containing a pointer and I wanted that to be expressed as a
relationship. JSONAPI doesn't allow that.

Thanks.
